MEMMODEL_CONSTRAINTS.md

```diff
@@ -156,6 +156,9 @@ may perceive in that statement.)
   * x86 (32-bit and 64-bit), at least Core2 and later
   * POWER (32-bit and 64-bit), unknown versions
   * SPARC (32-bit and 64-bit), v9 or later (none at present)
+* Systems programming languages have their own memory models:
+  * C/C++ (the C/C++11 or C/C++14 memory models)
+  * Rust (the LLVM memory model)
```

Comment:
I wouldn't put this under hardware: SAB is between language/compiler memory models and hardware memory models. Putting languages/compilers here is confusing.

I'd also only refer to C11/C++11 memory model, it hasn't substantially changed since then.

Also separate out Rust from LLVM.